catroid/src/main/java/org/catrobat/catroid/content/Project.java

```diff
@@ -526,4 +528,36 @@ public void checkForInvisibleSprites() {
 			scene.checkForInvisibleSprites();
 		}
 	}
+
+	public void checkIfSpriteNameEqualBackground(Activity activity) {
+		List<Sprite> spriteList =
+				new ArrayList<>(ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones());
+		for (int sprite = 1; sprite < spriteList.size(); ++sprite) {
+			if (spriteList.get(sprite).getName().matches("[\\s]*" + activity.getString(R.string.background) +
+					"[\\s]*")) {
+				setNewSpriteName(sprite, activity);
+				return;
+			}
+		}
+	}
+
+	public void setNewSpriteName(int sprite, Activity activity) {
+		List<Sprite> spriteList =
+				new ArrayList<>(ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones());
+		String backgroundRegex = activity.getString(R.string.background) + " \\([0-9]+\\)";
+		int maxValue = 0;
+		for (int spriteItr = 1; spriteItr < spriteList.size(); ++spriteItr) {
+			if (spriteList.get(spriteItr).getName().matches(backgroundRegex)) {
+				String tokens[] = spriteList.get(spriteItr).getName().split(" ");
+				int value = Integer.parseInt(tokens[1].replaceAll("[^0-9]", ""));
+				if (maxValue < value) {
+					maxValue = value;
+				}
+			}
+		}
+		int finalValue = maxValue + 1;
+		spriteList.get(sprite).setName(activity.getString(R.string.background) + " (" + finalValue +
+				")");
+	}
```

Comment:
In order to prevent duplicated code and make the code more maintainable, please use the already excising [UniqueNameProvider ](https://github.com/Catrobat/Catroid/blob/develop/catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/util/UniqueNameProvider.java) to get the new sprite name.

Reply:
Thank you, didn't see that function. Changes are pushed.